bindings/matrix-sdk-ffi/src/authentication_service.rs

```diff
@@ -97,4 +107,20 @@ impl AuthenticationService {
             None => Err(AuthenticationError::ClientMissing),
         }
     }
+
+    /// Get the homeserver login details from a client.
+    fn details_from_client(
+        &self,
+        client: &Arc<Client>,
+    ) -> Result<HomeserverLoginDetails, AuthenticationError> {
+        let homeserver = client.homeserver();
+        let authentication_issuer = client.authentication_issuer();
+        let supports_password_login =
+            client.supports_password_login().map_err(AuthenticationError::from)?;
```

Comment:
As these are all async it would be kind of nice to set them off together.
I had a go by removing the runtime blocks and marking them as `async`, but when I came to using `join!` the compiler complained like so:
```
3 | use futures_util::join;
  |     ^^^^^^^^^^^^^^^^^^ no `join` in the root
```
which seemed weird as it auto completes fine for me.

Reply:
This is because we don't activate `futures_util`'s default features (the macros are feature-gated): https://github.com/matrix-org/matrix-rust-sdk/blob/c20c23bc19ab5ccecd848cf4b97087fd1c50bb90/bindings/matrix-sdk-ffi/Cargo.toml#L23

Reply:
Ah ha that worked 🙏
Although turns out I actually needed the `join3` function anyway which was already available.